packages/obojobo-document-engine/src/scripts/oboeditor/components/sub-menu.js

```diff
@@ -27,8 +29,24 @@ class SubMenu extends React.Component {
 		EditorUtil.deletePage(pageId)
 	}
 
-	renamePage(pageId) {
-		const label = window.prompt('Enter the new title:') || pageId
+	showRenamePageModal(page) {
+		ModalUtil.show(
+			<Prompt
+				cancelOk
+				title="Rename Page"
+				message="Enter the new title for the page:"
+				value={page.label}
+				onConfirm={this.renamePage.bind(this, page.id)}
```

Comment:
need to pass label here or reference `this.label` is `renamePage`?

Reply:
Yes, this allows users to change the old name instead of just overwriting it with a new name
It also gives a reference to the user as to which page they are editing.